### PR TITLE
changed makefile to build rocm backend

### PR DESCRIPTION
**Pull request by adityaatluri**

### CODE REVIEW

**Comment by tqchen** [comment]
Lgtm, one thing is we have a dual build system(cmake for windows support) would be nice to add that as well

**Comment by adityaatluri** [comment]
ROCm is not yet supported on windows. Hence I didn't change it

**Comment by tqchen** [comment]
Ok, do you want to wait until we have a concrete runtime added before we merge this in?

**Comment by adityaatluri** [comment]
I think merging this will not create merge conflicts in future. It certainly makes life easier to merge small pieces of code rather than huge chunks of commits.

**Comment by tqchen** [comment]
Ok,  I think I can merge this in. But for runtime, it would be really nice to have the initial piece complete and have CI to verify the compilation correctness. Does Rcom have a compilation flag that support simulator like env, or Nvidia target? Since for now we do not yet have Amd card in our CI

**Comment by adityaatluri** [comment]
Can you explain a bit more on "Does Rcom have a compilation flag that support simulator like env, or Nvidia target? Since for now we do not yet have Amd card in our CI" ?

**Comment by tqchen** [comment]
We have a continuous integration service setup that every time a pull request is submitted, the code style checked, the project get rebuild, all the testcase are run to make sure that the change won't break anything to ensure the quality of code. Currently we have machines that setup with nvidia docker and nv card, so cuda opencl runtime is compiled and testcases are ran. We want to do the same thing for some extent for rocm backend, but we do not have machine with Amd card. That means either we Have to use nvidia card as alternative, or use some kinda of x86 simulated env, or we don't create testcase , but only verifies the code compiles for the very least

**Comment by adityaatluri** [comment]
Gotcha! Before you get AMD GPU integrated to CI machines, we can post the build and test output for each PR. Is it ok?

**Comment by tqchen** [comment]
Here is my proposal let us at least include RCom build system into the CI, so that at least it verifies:
   - 1. runtime compiles correctly with rcom lib dependency
   - 2. It generate the llvm necessary and possible binary necessary without failure
    -3. We should write test, but can skip the final execution step optionally when AMD card is not available.
        - This can be done via tvm.module.enabled("rocm") to check if runtime module is enabled and skip the result verification steps, but we should write tests that verifies the result, so you can check if they run correctly on your build system. c.f. https://github.com/dmlc/tvm/blob/master/tests/python/integration/test_gemm.py#L66

Basically we can verify the compilation phase as a cross compilation step

**Comment by adityaatluri** [comment]
Sounds great.
